### ISSUE TITLE
Issue with recent `initval` changes

### ISSUE THREAD

**Issue by ricardoV94** [pymc-devs/pymc#4737]
I found some issues with the recent `initval` changes in https://github.com/pymc-devs/pymc3/pull/4729, as it doesn't respect the parents initvals.

This snippet often leads to a `ValueError`:

```python
with pm.Model() as model:
    mu = pm.Normal('mu', initval=100)
    alpha = pm.HalfNormal('alpha', initval=100, transform=None)
    value = pm.NegativeBinomial('value', mu=mu, alpha=alpha)

model.initial_values
# {mu: array(100., dtype=float32),
#  alpha: array(100., dtype=float32),
#  value: array(1)}
```

When it doesn't fail, initval is still far from the expected ~ 100:
```python
pm.NegativeBinomial.dist(mu=100, alpha=100).eval()
# array(98)
```

_Originally posted by @ricardoV94 in https://github.com/pymc-devs/pymc3/issues/4734#issuecomment-854495715_

**Comment by ricardoV94** [comment]
Also a model where `pm.Flat` is used as input will now raise an error unless an initval is set for those variables:

```python
with pm.Model() as m:
    x = pm.Flat('x')
    y = pm.Normal('y', x, 1)    
```
```
NotImplementedError: Cannot sample from flat variable
```

**Comment by michaelosthege** [comment]
closed by #4738
